active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,320 @@
+
+XAML
+```XAML
+<NumberBox Header="Quantity"
+    Value="{x:Bind Path=ViewModel.NumberBoxValue, Mode=TwoWay}"
+    ValidationMode="InvalidInputOverwritten" />
+```
+
+Setting `ValidationMode` to `Disabled` allows custom input validation to be configured.  
+
+### Formatting input 
+
+[Number formatting](https://docs.microsoft.com/en-us/uwp/api/windows.globalization.numberformatting) can be used to format the value of a Numberbox by configuring an instance of a formatting class and assigning it to the `NumberFormatter` property. Decimal, currency, percent, and significant figures are few of the number formatting classes available. 
+
+Here is an example of using DecimalFormatter to format a NumberBox's value to have one integer digit and two fraction digits:  
+
+XAML
+```XAML
+<NumberBox  x:Name="FormattedNumberBox"
+Value="{x:Bind Path=ViewModel.NumberBoxValue, Mode=TwoWay}" />
+```
+
+C#
+```C#
+private void SetNumberBoxNumberFormatter()
+{
+    DecimalFormatter formatter = new DecimalFormatter();
+    formatter.IntegerDigits = 1;
+    formatter.FractionDigits = 2;
+    FormattedNumberBox.NumberFormatter = formatter;
+}
+```
+
+![A NumberBox showign a value of 0.00.](images/numberbox-formatted.PNG)
+
+### Enabling hyper scroll 
+
+Setting `IsHyperScrollEnabled` to true enables the value of NumberBox to be incremented or decremented by scrolling when a NumberBox has focus and is being hovered over. 
+
+XAML
+```XAML
+<NumberBox Value="{x:Bind Path=ViewModel.NumberBoxValue, Mode=TwoWay}"
+    IsHyperScrollEnabled="True" />
+```
+
+## Remarks
+
+### Input Scope
+
+`Number` will be used for the [input scope](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Input.InputScopeNameValue). This input scope is intended for working with digits 0-9. This may be overwritten but alternative InputScope types will not be explicitly supported. 
+
+### Not a Number
+
+When a NumberBox is cleared of input, `Value` will be set to `NaN` to indicate no numerical value is present. 
+
+## Recommendations
+
+* `Text` and `Value` make it easy to capture the value of a NumberBox as a String or as a Double without needing to convert the value bewteen types. When programmatically altering the value of a NumberBox, it is recommended to do so through the `Value` property. `Value` will overwrite `Text` in initial set up. After the initial set up, changes to one will be progrogated to the other, but consistently making programmatic changes through `Value` helps avoid any conceptual misunderstanding that NumberBox will accept non-numeric characters through `Text`.  
+
+## API Notes
+
+### Notable Properties  
+
+| Name | Description |
+|:-:|:--|
+| StepFrequency | Gets or sets the value part of a value range that steps should be created for. |
+
+## API Details
+<!-- todo: Missing some APIs in here, like TemplatSettings class and event args classes -->
+
+```c++ 
+enum NumberBoxSpinButtonPlacementMode
```

Comment:
Is the "Spin Button" multiple buttons actually? Does it really spin? (Im not familiar with the surface so I was surprised on this name)

Reply:
Spin Button is the traditional windows and HTML name for the Up and Down buttons.

Reply:
Spin Button is the traditional HTML and Win32 name given to the Up and Down Buttons.

Reply:
Discussion: Yes, "spin button" is prior art for the buttons in a spinner control. (NumberBox is a "spinner" control, and sets [AutomationControlType.Spinner](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Automation.Peers.AutomationControlType#fields) as the automation type.

NumberBox isn't being called "spinner" though because this additionally supports calculation.

FabricUI precedence is "up down buttons"

Reply:
Also a precedent in web.

We missed the recommendation in the response here.

Recommendation: Leave as is.